tests/unit/Service/EncodingServiceTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace OCA\Text\Service;
+
+use Test\TestCase;
+
+class LabelServiceTest extends TestCase {
+	private $encodingService;
+
+	protected function setUp(): void {
+		parent::setUp();
+		$this->encodingService = new EncodingService();
+	}
+
+	/**
+	 * Attempt to decode the file using the default decoding order.
+	 * For files with encodings not included in the COMMON_ENCODINGS array encoding to UTF-8 will fail.
+	 * We then do the conversion again after setting the mb_detect_order value: all conversions should succeed.
+	 * @dataProvider dataFileEncodings
+	 */
+	public function testDefault(string $file, string $encoding) {
+		$utf8_string = $this->encodingService->encodeToUtf8(file_get_contents($file));
+
+		// If encoding is not part of the default encodings we can expect it to fail
+		// It might still succeed because encoding detection is not precise.
+		if ($utf8_string || $this->isSupportedEncoding($encoding)) {
+			$this->assertNotNull($utf8_string);
+			$this->assertNotFalse(mb_detect_encoding($utf8_string, 'UTF-8', true));
+		}
+
+		$original_order = mb_detect_order();
+		$this->assertNotFalse(mb_detect_order($encoding));
+
+		$utf8_string = $this->encodingService->encodeToUtf8(file_get_contents($file));
+		$this->assertNotNull($utf8_string);
+		$this->assertNotFalse(mb_detect_encoding($utf8_string, 'UTF-8', true));
+
+		mb_detect_order($original_order);
+	}
+
+	/**
+	 * If the encoding is in the list of common encodings we should be able to detect an encoding (it might not be the
+     * correct encoding due to detection inaccuracies). If not, add the encoding to mb_detect_order.
+	 * @dataProvider dataFileEncodings
+	 */
+	public function testDetection(string $file, string $encoding) {
+		$detected_encoding = $this->encodingService->detectEncoding(file_get_contents($file));
+		if ($this->isSupportedEncoding($encoding)) {
+			$this->assertNotNull($detected_encoding);
+		}
+
+		$original_order = mb_detect_order();
+		$this->assertNotFalse(mb_detect_order($encoding));
+
+		$detected_encoding = $this->encodingService->detectEncoding(file_get_contents($file));
+		$this->assertEquals($encoding, $detected_encoding);
+
+		mb_detect_order($original_order);
```

Comment:
Small nitpick on the code style 😉 I guess we currently don't run php-cs-fixer on the tests directory.
```suggestion
		$detectedEncoding = $this->encodingService->detectEncoding(file_get_contents($file));
		if ($this->isSupportedEncoding($encoding)) {
			$this->assertNotNull($detectedEncoding);
		}

		$originalOrder = mb_detect_order();
		$this->assertNotFalse(mb_detect_order($encoding));

		$detectedEncoding = $this->encodingService->detectEncoding(file_get_contents($file));
		$this->assertEquals($encoding, $detectedEncoding);

		mb_detect_order($originalOrder);
```